File: Problems/Leetcode/3001-4000/3101-3200/3200.cpp
#include <bits/stdc++.h>

using namespace std;

static constexpr int MOD = 1e9 + 7;

int calc(int x, int y) {
    int cnt = 1;

    while (true) {
        if (x >= cnt) {
            x -= cnt++;
        } else {
            break;
        }
        if (y >= cnt) {
            y -= cnt++;
        } else {
            break;
        }
    }
    return cnt - 1;
}

class Solution {
public:
    int maxHeightOfTriangle(int red, int blue) { return max(calc(red, blue), calc(blue, red)); }
};

int main() {
    Solution sol;

    return 0;
}

File: Problems/Leetcode/3001-4000/3101-3200/3200.py
from itertools import count
from math import isqrt, sqrt


class Solution:
    def maxHeightOfTriangle(self, red: int, blue: int) -> int:  # type: ignore
        """
        枚举
        """
        # cnt = [0, 0]
        # for i in count(1):
        #     cnt[i % 2] += i
        #     if (cnt[0] > red or cnt[1] > blue) and (cnt[0] > blue or cnt[1] > red):
        #         return i - 1

        """
        Math
        """

        def f(n: int, m: int) -> int:
            odd = isqrt(n)
            even = int((sqrt(m * 4 + 1) - 1) / 2)
            return even * 2 + 1 if odd > even else odd * 2

        return max(f(red, blue), f(blue, red))
